Hazelcast.Net/Hazelcast.Client.Spi/ClientPartitionService.cs

```diff
@@ -48,9 +48,22 @@ public ClientPartitionService(HazelcastClient client)
 
         public Address GetPartitionOwner(int partitionId)
         {
-            Address rtn;
-            _partitions.TryGetValue(partitionId, out rtn);
-            return rtn;
+            Address partitionOwner = null;
+            while (_live.Get() && !_partitions.TryGetValue(partitionId, out partitionOwner))
+            {
+                Logger.Finest("Address of a partition cannot be null. Retrying to get it...");
```

Comment:
Printing in a busy loop may result too many messages, do we really want to do it?

Reply:
I think instead of this we can reuse the same code that we use in GetPartitions. 
```
if (_partitionCount == 0)
{
         GetPartitionsBlocking();
}
_partitions.TryGetValue(partitionId, out partitionOwner)  <=== this is guaranteed to work
```

Reply:
GetPartitionsBlocking(); blocks for 10 sec which is too much for here. I'll just add a sleep